test/p2p_c_version/answerer.c

```diff
@@ -0,0 +1,323 @@
+                        rtcAddRemoteCandidate(peer->pc, candidate, "1");
+                        free(candidate);
+
+                  }else {
+                        printf("Error reading line\n");
+                        break;
+                  }
+
+			break;
+		}
+		case 3: {
+			// Send Message
+                  if(!peer->connected){
+				printf("** Channel is not Open **");
+				break;
+			}
+			printf("[Message]: ");
+                  char* message = NULL;
+			size_t message_size = 0;
+                  int c;
+                  while ((c = getchar()) != '\n' && c != EOF) { }
+                  if(getline(&message, &message_size, stdin)){
+                        rtcSendMessage(peer->dc, message, -1);
+                        free(message);
+                  }else {
+                        printf("Error reading line\n");
+                        break;
+                  }
+			break;
+		}
+		case 4: {
+			// Connection Info
+                  if(!peer->connected){
+				printf("** Channel is not Open **");
+				break;
+			}
+                  char buffer[256];
+                  if (rtcGetLocalAddress(peer->pc, buffer, 256) >= 0)
+                        printf("Local address 1:  %s\n", buffer);
+                  if (rtcGetRemoteAddress(peer->pc, buffer, 256) >= 0)
+                        printf("Remote address 1: %s\n", buffer);
+
+			else
+				printf("Could not get Candidate Pair Info\n");
+			break;
+		}
+		default: {
+			printf("** Invalid Command **");
+			break;
+		}
+		}
+	}
+
+
+	deletePeer(peer);
+	return 0;
+}
+
+
+static void descriptionCallback(const char *sdp, const char *type, void *ptr) {
+	// Peer *peer = (Peer *)ptr;
+	printf("Description %s:\n%s\n", "answerer", sdp);
+}
+
+static void candidateCallback(const char *cand, const char *mid, void *ptr) {
+	// Peer *peer = (Peer *)ptr;
+	printf("Candidate %s: %s\n", "answerer", cand);
+
+}
+
+static void stateChangeCallback(rtcState state, void *ptr) {
+	Peer *peer = (Peer *)ptr;
+	peer->state = state;
+	printf("State %s: %s\n", "answerer", state_print(state));
+}
+
+static void gatheringStateCallback(rtcGatheringState state, void *ptr) {
+	Peer *peer = (Peer *)ptr;
+	peer->gatheringState = state;
+	printf("Gathering state %s: %s\n", "answerer", rtcGatheringState_print(state));
+}
+
+
+// static void openCallback(void *ptr) {
+// 	Peer *peer = (Peer *)ptr;
+// 	peer->connected = true;
+//       printf("Reached this openCallback\n");
+//       // char buffer[256];
+// 	// if (rtcGetDataChannelLabel(peer->dc, buffer, 256) >= 0)
+// 	// 	printf("DataChannel %s: Received with label \"%s\"\n","answerer", buffer);
+//       //
+//
+// }
+
+static void closedCallback(void *ptr) {
+	Peer *peer = (Peer *)ptr;
+	peer->connected = false;
+
+      // char buffer[256];
+
+      // if (rtcGetDataChannelLabel(peer->dc, buffer, 256) >= 0)
+      //       printf("DataChannel %s: Received with label \"%s\"\n","answerer", buffer);
```

Comment:
What was this for ? :thinking:

Reply:
I was experimenting with some this to see whether I could print a close message or something like that but did not work jajajajja